sql/src/main/java/io/crate/executor/transport/TransportFetchNodeAction.java

```diff
@@ -101,7 +101,7 @@ public void nodeOperation(final NodeFetchRequest request,
             statsTables.operationStarted(request.fetchPhaseId(), request.jobId(), "fetch");
 
             // nothing to fetch, just close
-            if (request.toFetch() == null) {
+            if (request.toFetch() == null && request.isCloseContext()) {
```

Comment:
Normally we don't send requests where there is nothing to fetch and isCloseContext is false:
Check code if FetchProjector line 182:

`
if (toFetch.isEmpty() && !isLast) {
     remainingRequests.decrementAndGet();
 } else {
     final String nodeId = entry.getKey();
     ListenableFuture<IntObjectMap<? extends Bucket>> future = fetchOperation.fetch(nodeId, toFetch, isLast);
`